ReactWindows/ReactNative.Net46/Bridge/DispatcherHelpers.cs

```diff
@@ -17,12 +18,47 @@ public static void AssertOnDispatcher()
 
         public static bool IsOnDispatcher()
         {
-            return Thread.CurrentThread == Dispatcher.CurrentDispatcher.Thread;
+            var dispatcher = Application.Current != null
+                ? Application.Current.Dispatcher
+                : Dispatcher.CurrentDispatcher;
+
```

Comment:
From our debug session yesterday it seems like it doesn't produce the correct behavior.

---
In reply to: [87300281](https://github.com/ReactWindows/react-native-windows/pull/868#discussion_r87300281) [](ancestors = 87300281)